src/parser/semantic_check.c

```diff
@@ -13840,7 +13887,31 @@ pt_check_class_eq (PARSER_CONTEXT * parser, PT_NODE * p, PT_NODE * q)
 	{
 	  return 1;
 	}
-      break;
```

Comment:
Not related to the issue. How about putting the following condition first?
```
if (p->info.name.spec_id != q->info.name.spec_id)
	{
	  return 1;
	}
```

Reply:
I agree. I will change it.